### PR TITLE
[rust] Make it easier to test rust client generator

### PR DESCRIPTION
**Pull request by bjgill**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language. @frol @farcaller @richardwhiuk 

### Description of the PR

As we discovered in #3463, there are various bits of the rust client generator that are currently untested. This PR adds tests for various generator-specific types, and also files.

Follows the example of the `rust-server` generator for making it easy to add new test specs. This should make it easier for future contributors to make sure that their contributions are tested.

### CODE REVIEW

**Comment by bjgill** [comment]
Note for reviewers: I'd suggest reviewing this by commits, rather than trying to do the whole thing together. The bulk of the changes are the four new samples, which are largely uninteresting...

File: bin/rust-petstore.sh [hunk 1]
```diff
@@ -27,6 +27,21 @@ fi
...
-java ${JAVA_OPTS} -jar ${executable} ${ags}
+for spec_path in \
+  modules/openapi-generator/src/test/resources/2_0/petstore.yaml \
+  modules/openapi-generator/src/test/resources/3_0/rust/rust-test.yaml \
+  modules/openapi-generator/src/test/resources/2_0/fileResponseTest.json\
+  ; do
+  spec=$(basename "$spec_path" | sed 's/.yaml//' | sed 's/.json//' )
+
+  for library in hyper reqwest; do
+    ags="generate --template-dir modules/openapi-generator/src/main/resources/rust
```
Comment (richardwhiuk) [inline bin/rust-petstore.sh]: You probably want `args` not `ags`, and you probably want to check the return code.

Comment (bjgill) [reply bin/rust-petstore.sh]: I'm not sure - `rust-server` has `args`, but everything else appears to use `ags`. I don't think it matters much, though, so will switch.

Good catch - we can no longer rely on implicitly returning a non-zero code on failure. I'll fix that.

**Comment by bjgill** [comment]
OK. I've rebased cleanly onto master. Assuming this builds fine, I'll merge.

**Comment by wing328** [comment]
@bjgill  thanks for the PR, which has been included in v4.2.0 release: https://twitter.com/oas_generator/status/1189824932345069569

**Pull request merged**